### PR TITLE
feat: Move from hcl to yaml

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

File: docs/cli/docker.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ docker run \
   -v <PATH TO CACHE>/.cq:/data/.cq \
   -v <PATH TO CONFIG>/config.hcl:/config/config.hcl \
```
Comment (disq) [inline docs/cli/docker.md:63]: Missed here

File: docs/configuration/environment-variable-substitution.md [hunk 2]
```diff
@@ -1,34 +1,30 @@
...
-```hcl
-cloudquery {
-  provider "aws" {
-    source  = ""
-    version = "${AWS_VERSION}"
-  }
+```yml
+cloudquery:
+  providers:
+    - name: "aws"
```
Comment (disq) [inline docs/configuration/environment-variable-substitution.md:16]: I would remove quotes as much as possible

File: docs/configuration/overview.md [hunk 3]
```diff
@@ -1,30 +1,25 @@
...
-    username = "postgres"
-    password = "pass"
-    host = "localhost"
-    port = 5432
-    database = "postgres"
-    sslmode = "disable"
+```yml
+cloudquery:
+  connection:
+    type: "postgres"
```
Comment (disq) [inline docs/configuration/overview.md:13]: Remove all quotes

**Comment by yevgenypats** [comment]
The cli subsection will be auto generated by this command:

https://github.com/cloudquery/cloudquery/pull/989

File: docs/configuration/environment-variable-substitution.md (hunk 2 above)
Comment (yevgenypats) [reply docs/configuration/environment-variable-substitution.md:16]: why? because it's cleaner?

Comment (disq) [reply docs/configuration/environment-variable-substitution.md]: yes

File: docs/cli/rate_limiting.md [hunk 4]
```diff
@@ -5,34 +5,31 @@ You can configure CloudQuery to limit the amount of resources fetched in paralle
...
 
-```hcl
-provider "aws" {
-  configuration {
-    aws_debug = false
-  }
+```yml
+providers:
+  - name: aws
+    configuration:
```
Comment (disq) [inline docs/cli/rate_limiting.md:25]: I don't think we have a configuration block now

Comment (yevgenypats) [reply docs/cli/rate_limiting.md:25]: So what we have instead ?

Comment (yevgenypats) [reply docs/cli/rate_limiting.md:25]: just raw config?

File: docs/getting-started/_getting-started-spawn-section.mdx [hunk 5]
```diff
@@ -22,25 +22,23 @@ password `pass`. [After installing docker](https://www.docker.com/get-started),
...
+cloudquery:
   ...
   ...
 
-  connection {
-    dsn = "host=localhost user=postgres password=pass database=postgres port=5432"
-  }
-}
+  connection:
+    dsn: "host=localhost user=postgres password=pass database=postgres port=5432"
```
Comment (bbernays) [inline docs/getting-started/_getting-started-spawn-section.mdx:41]: This should be changed to new format

File: docs/getting-started/getting-started-with-aws.mdx [hunk 6]
```diff
@@ -135,18 +135,16 @@ Then, you can either export the `AWS_PROFILE` environment variable:
...
-provider "aws" {
-  configuration {
-    accounts "<account_alias>" {
-      local_profile = "myprofile"
-    }
+```yml title="config.yml"
+providers:
+  - name: "aws"
+    accounts:
+      - name: "<account_alias>"
```
Comment (bbernays) [inline docs/getting-started/getting-started-with-aws.mdx:144]: In other places we refer to this as `<YOUR ID>`

File: docs/tutorials/assume_role.mdx [hunk 7]
```diff
@@ -130,39 +130,30 @@ aws iam attach-user-policy --user-name <SourceAccountUserName> --policy-arn "arn
...
   ...
   
-  provider "aws" {
-    source = "cloudquery/aws"
-    version = "latest"
-  }
+  providers:
+    - name: "aws"
+      source: "cloudquery/aws"
+      version: "latest"
```
Comment (bbernays) [inline docs/tutorials/assume_role.mdx:143]: Do we ever document all of the attributes in the `providers` block?

File: docs/tutorials/assume_role.mdx [hunk 8]
```diff
@@ -130,39 +130,30 @@ aws iam attach-user-policy --user-name <SourceAccountUserName> --policy-arn "arn
...
 
-  connection {
-    dsn = "<PostgreSQL_DSN>"
-  }
-}
+  connection:
+    dsn: "<PostgreSQL_DSN>"
```
Comment (bbernays) [inline docs/tutorials/assume_role.mdx:146]: Should use new format

File: docs/tutorials/assume_role.mdx (hunk 7 above)
Comment (yevgenypats) [reply docs/tutorials/assume_role.mdx:143]: I think the documentation should go in json schema validation eventually

**Review by disq — APPROVED** [review]

File: docs/tutorials/assume_role.mdx (hunk 7 above)
Comment (bbernays) [reply docs/tutorials/assume_role.mdx:143]: But isn't that the code? How are we going to represent it in the docs?

**Pull request merged**